extensions/roulette/database/redis_db.py

```diff
@@ -0,0 +1,95 @@
+import logging
+
+from Redis.redis_config import get_redis
+from ..config import config
+from .action import fetch_timeout_roles
+
+from datetime import datetime, timedelta
+from discord import Member, Message
+from typing import Set
+
+_redis_client = get_redis()
+
+logger = logging.getLogger("roulette.database")
+
+
+# Mute functions
+# TODO: Check for successful update
+async def _upload_timeout(duration: timedelta,
+                          message: Message,
+                          target: Member) -> int:
+
+    key = config.redis_key_const() + str(message.guild.id)
+
+    time_unmute = datetime.now() + duration
+    posix_time_unmute = int(time_unmute.timestamp())
+
+    resp = _redis_client.zadd(key, {target.id: posix_time_unmute})
+    return resp
+
+
+async def _apply_timeout_roles(target, duration_label):
+    timeout_ids = config.timeout_roles()
+
+    for timeout_id in timeout_ids:
+        role = target.guild.get_role(int(timeout_id))
+
+        if role:
+            await target.add_roles(role, reason=f"Adding role for {duration_label} timeout")
+
+
+async def add_timeout(target, duration_label, duration, message):
+    await _upload_timeout(duration, message, target)
+
+
+# Unmute functions
+async def _get_guild_ids() -> Set[int]:
+    cursor = 0
+    pattern = config.redis_key_const() + '*'
+
+    guild_keys = set()
+    guild_ids = set()
+
+    while True:
+        cursor, partial_keys = _redis_client.scan(cursor=cursor, match=pattern)
+        guild_keys.update(partial_keys)
+
+        if cursor == 0:
+            break
+
+    for key in guild_keys:
+        guild_ids.add(int(key[22:]))
+
+    return guild_ids
+
+
+async def _fetch_candidates(guild_id) -> Set[int]:
```

Comment:
nit: `candidates` is a bit vague here. These specifically are muted users who may be unmutable.